Assets/Scripts/Player/PlayerAtack.cs

```diff
@@ -0,0 +1,40 @@
+using System.Linq;
+using UnityEngine;
+using System;
+
+public class PlayerAtack : MonoBehaviour
+{
+    [SerializeField] int dmg = 1;
+    [SerializeField] float force = 2f;
+    [SerializeField] float range = .8f;
+    [SerializeField] private LayerMask enemyLayer;
+    [SerializeField] GameObject rock;
+
+    public void Attack()
+    {
+        var enemies = Physics2D.OverlapCircleAll(transform.position, range, enemyLayer).Where(e => e.CompareTag("Enemy"));
+        foreach (var enemy in enemies)
+        {
+            enemy.GetComponent<Health>().Damage(dmg);
+        }
+    }
+
+    public void Throw(float powerPercentage)
+    {
+        if (powerPercentage < 0 || powerPercentage > 1) throw new ArgumentOutOfRangeException("you can only throw with percentage power between 0% and 100%!!");
```

Comment:
I dont know how I feel about throwing exceptions in cases like this - it could break the flow of the game and is not really necessary in this context, maybe just do nothing and log the issue to the console instead?

Reply:
done